pkg/cluster/internal/providers/podman/images.go

```diff
@@ -83,10 +83,25 @@ func pull(logger log.Logger, image string, retries int) error {
 
 // sanitizeImage is a helper to return human readable image name and
 // the podman pullable image name from the provided image
-func sanitizeImage(image string) (string, string) {
+func sanitizeImage(image string) (friendlyImageName, pullImageName string) {
```

Comment:
Only dealt with our actual situation, and users may upload the node image to their own private registry.

Did not judge scenarios such as users using `docker.io/library/xxx:vvv`

Reply:
we should probably leave a TODO then, because in the case of someone using library someday (??) this will break

Reply:
also what about a custom registry? `foo.bar/baz:quux` ?

Reply:
Addressed, please see test cases.